modules/dists/PrivateDist.chpl

```diff
@@ -151,15 +167,16 @@ private proc checkCanMakeDefaultValue(type eltType) param {
 }
 
 class PrivateArr: BaseRectangularArr {
+  param initElts: bool;
```

Comment:
I think it's great that this change fixes the problem, but adding a `param` field here might have unintended consequences. You might find that creating a `PrivateArr` in two different ways will get it to have two different types. Will assignment work between these types? Comparison? I am worried that these issues will not be discovered due to light testing on `PrivateArr`.

What would make more sense to me would be to tell the compiler (somehow, maybe with a pragma setting a flag) not to worry about the default initialization inside of the `if defaultInitDataOnPrivatize` and to pair that with a compilation error in the main `proc init` based on `initElts`. (Actually don't we already get an error in that case because of the pattern there?).

AFAIK if you mark the function or the variable with `pragma "unsafe"` it will hide these errors.

Reply:
OK, I'll give `pragma "unsafe"` a try for now. In the future (say after the release), wouldn't it be better to just use `param initElts` and just focus on adding more testing to cover the potential cases that you mention?

Reply:
I think it's better not to change the type if we don't absolutely have to since it impacts the user interface for PrivateArr.